### DISCUSSION TITLE
How to have Matter and Thread on HA using Z2M with Sonoff dongle E ?

### DISCUSSION THREAD

**Discussion by kevincaradant** [Koenkk/zigbee2mqtt#21556] [General]
Hello everyone,

Maybe my question might seem obvious, but now I'm a bit lost and not sure how to proceed :(. I hope I'm asking it in the right place on the Z2M topic.

I'm a bit confused with Zigbee / Matter / Thread / Home Assistant / Open Thread Border Router / Matter Server.

Basically, today I have a device (Nuki Smart Lock) that I can connect using Matter. When I try to connect, it asks me to scan a QR code and use my home automation solution (in my case, Home Assistant).
I'm currently running with CC1352P-2 and 100 devices on Home Assistant.

SO to add this Matter Device. I've also bought (perhaps unnecessarily, you tell me ? ) a Sonoff dongle E because I wanted to have a Matter device. ( As I don't have a Google or Apple Hub). I followed this tutorial 
![image](https://github.com/Koenkk/zigbee2mqtt/assets/6093175/a4d70c1c-f75c-4fcd-8bd8-919500aff84f)
and selected "multi pan". Should I actually choose "Open Thread" instead?

Essentially, I want this Sonoff to be a Matter router because alongside it, I have a Zigbee 3.0 coordinator Texas Instruments LAUNCHXL-CC1352P-2. Can I connect both to my NUC (server)? Is there any benefit? I don't want to replace my CC1352P-2 with the new Sonoff for the Zigbee part. I just want to add my Sonoff as a Matter router.

In parallel, I've also installed a Matter emulator (https://github.com/home-assistant-libs/python-matter-server). And I've installed the Matter add-on (BETA) in Home Assistant, setting my server itself as the address. Matter seems to be okay.

And here's where I'm lost. There's also the story of Thread using Matter with this plugin in Home Assistant "Open Thread Border Router", but do I need to go through Home Assistant or Zigbee2Mqtt? It asks me for a REST API address.

Finally, to confuse me a bit more, I came across this link: https://github.com/antonio-gabriele/zigbee-matter-bridge.

So now, I don't know what to think anymore. I feel like I have all the pieces but don't know how to put the puzzle together :)

To summarize:

```
- I use HA with Z2M and a coordinator CC1352P-2.

- I have a Nuki that I want to connect using Matter.

- I have a Sonoff dongle for the Matter/Thread part that I don't know how to use as a router.

- So, I want my Nuki in HA using Matter.
```

Thanks in advance for shedding some light on my confusion.

**Comment by kevincaradant** [comment]
OK after hours of tries,
I succesfully udpated the firmware using multi pan because with the other one doesn't work correctly.

Then I setup https://github.com/b2un0/silabs-multipan-docker which allow to me to use Open Thread Border Router with the Thread Add On.

So it's seems OK to have Thread / Matter on HA using Sonoff dongle E. 

I'm currently using it on channel 15 for Thread

My others Zigbee devices are on channel 20 but I'm using an other coordinator CC1352P-2 so I think it's not related between them. 

Now HA can see my Nuki but when I want to linked them from the App HA, I got an issue 'Impossible to connect the device to thread HA'. 
I will open an issue on HA for that.
